### PR TITLE
Warn on invalid input

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### What does this implement/fix? Explain your changes.
There were places in menu which was formerly ignored if user gives invalid input. Some of them (listed below) are now checked before start generating graphs.
+ all 0 and 1 options are now restricted to only `0` and `1`
+ `file_name` now can not be empty
+ `output_format` should be inside the valid range
+ `engine` should be inside the valid range
+ `vertices`, `number_of_files`, `min_edge`, `max_edge` are now restricted to positive integers

#### Any other comments?
Checking if `max_edge` > `min_edge` or `max_weight` > `min_weight` seems to be handled in a different way. If it's necessary we should do it in another PR.

### CODE REVIEW

File: pyrgg/functions.py [hunk 1]
```diff
@@ -62,7 +62,33 @@ def is_float(input_number):
         return True if decimalpart else False
 
 
-def convert_str_to_number(string):
+def handle_string(x):
+    """
+    Handle string.
```
Comment (sepandhaghighi) [inline pyrgg/functions.py:67]: Add more description to this docstring

File: pyrgg/functions.py [hunk 2]
```diff
@@ -62,7 +62,33 @@ def is_float(input_number):
...
+    :return: input string
+    """
+    if x == "":
+        raise ValueError
+    return x
+
+
+def handle_pos_int(x):
+    """
+    Handle int.
```
Comment (sepandhaghighi) [inline pyrgg/functions.py:80]: Add more description to this docstring

File: pyrgg/params.py [hunk 3]
```diff
@@ -10,8 +10,8 @@
         1- Pyrgg engine
         """
         )],
-    2: ["file_name", "- File Name : "],
-    3: ["number_of_files", "- Number of Files : "],
+    2: ["file_name", "- File Name (not empty) : "],
+    3: ["number_of_files", "- Number of Files (>=0) : "],
```
Comment (sepandhaghighi) [inline pyrgg/params.py:14]: `>=0` or `>=1` ?

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: pyrgg/params.py (hunk 3 above)
Comment (sadrasabouri) [reply pyrgg/params.py:14]: I think it makes more sense it it's `>=0`. Image the case that you just wanted to make configs.

File: pyrgg/params.py [hunk 4]
```diff
@@ -56,11 +56,11 @@
...
 PYRGG_ENGINE_PARAMS = {
-    1: ["vertices", "- Vertices Number : "],
-    2: ["min_edge", "- Min Edge Number (Connected to Each Vertex) : "],
-    3: ["max_edge", "- Max Edge Number (Connected to Each Vertex) : "],
-    4: ["min_weight", "- Min Weight : "],
-    5: ["max_weight", "- Max Weight : "],
+    1: ["vertices", "- Vertices Number (>=0) : "],
+    2: ["min_edge", "- Min Edge Number - Connected to Each Vertex (>=0) : "],
+    3: ["max_edge", "- Max Edge Number - Connected to Each Vertex (>=0) : "],
+    4: ["min_weight", "- Min Weight (>=0) : "],
```
Comment (sepandhaghighi) [inline pyrgg/params.py:62]: It's wrong, `min_weight` can be negative!

File: pyrgg/params.py [hunk 5]
```diff
@@ -56,11 +56,11 @@
...
+    5: ["max_weight", "- Max Weight (>=0) : "],
```
Comment (sepandhaghighi) [inline pyrgg/params.py:63]: It's wrong, `max_weight` can be negative!

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

**Review by sadrasabouri — COMMENTED** [review]
I double-checked it. It's OK.

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**